src/names/preludes.md

```diff
@@ -0,0 +1,144 @@
+# Preludes
+
+A *prelude* is a collection of names that are automatically brought into scope
+of every module in a crate.
```

Comment:
AFAIK, originally "prelude" meant simply a piece of code inserted by the compiler at the start of the crate, like
```rust
#[prelude_import]
use std::prelude::v1::*;
#[macro_use]
extern crate std;
```
, not a set of names.

Then I used it as a temporary name / analogy for built-in types (["language prelude"](https://github.com/rust-lang/rust/pull/32131)) and extern crates (["extern prelude"](https://github.com/rust-lang/rust/pull/49789)), because they behaved similarly to the actual prelude import, and then it stuck.

I still have no idea whether the name "prelude" sounds reasonable in the meaning "set of names" to native speakers, or to users of [languages](https://hackage.haskell.org/package/base-4.14.1.0/docs/Prelude.html) from which this term originally came.

Reply:
I'm personally comfortable with it. I think the behavior is similar enough ("importing these things into every module"). I think it will help later when talking about behaviors in how names are resolved.

Reply:
> I still have no idea whether the name "prelude" sounds reasonable in the meaning "set of names" to native speakers, or to users of languages from which this term originally came.

As a native speaker, I think of "prelude" as a jargon term the same way "macro" or "scope" is. Those words all have meanings outside programming, but the connection to the jargon term is tenuous at best. I think "prelude" is useful as an existing jargon term that's used in other languages, though.